pkg/cmd/server/origin/run_components.go

```diff
@@ -291,8 +291,12 @@ func (c *MasterConfig) RunBuildPodController() {
 func (c *MasterConfig) RunBuildImageChangeTriggerController() {
 	bcClient, _ := c.BuildImageChangeTriggerControllerClients()
 	bcInstantiator := buildclient.NewOSClientBuildConfigInstantiatorClient(bcClient)
-	factory := buildcontrollerfactory.ImageChangeControllerFactory{Client: bcClient, BuildConfigInstantiator: bcInstantiator}
-	factory.Create().Run()
+	bcIndex := &oscache.StoreToBuildConfigListerImpl{c.Informers.BuildConfigs().Indexer()}
+	bcIndexSynced := c.Informers.BuildConfigs().Informer().HasSynced
+	factory := buildcontrollerfactory.ImageChangeControllerFactory{Client: bcClient, BuildConfigInstantiator: bcInstantiator, BuildConfigIndex: bcIndex, BuildConfigIndexSynced: bcIndexSynced}
+	go func() {
+		factory.Create().Run()
+	}()
```

Comment:
@smarterclayton 
Every other controller that's being created is calling its own run method after creating the controller.  Are you saying these are all wrong?

https://github.com/openshift/origin/blob/master/pkg/cmd/server/origin/run_components.go#L88-L89
https://github.com/openshift/origin/blob/master/pkg/cmd/server/origin/run_components.go#L108
https://github.com/openshift/origin/blob/master/pkg/cmd/server/origin/run_components.go#L158
https://github.com/openshift/origin/blob/master/pkg/cmd/server/origin/run_components.go#L176
https://github.com/openshift/origin/blob/master/pkg/cmd/server/origin/run_components.go#L337
https://github.com/openshift/origin/blob/master/pkg/cmd/server/origin/run_components.go#L358